crates/sdk/src/connector.rs

```diff
@@ -135,22 +135,21 @@ pub trait Connector: Send {
 
 /// Connectors are set up by values that implement this trait.
 ///
-/// It provides a method for parsing configuration, and another for initializing
-/// state.
+/// It provides a method for parsing configuration, and another for initializing state.
 ///
 /// See [`Connector`] for further details.
 #[async_trait]
-pub trait ConnectorSetup {
+pub trait ConnectorSetup: Send + Sync + 'static {
     type Connector: Connector;
 
     /// Validate the configuration provided by the user, returning a configuration error or a
-    /// validated [`Connector::Configuration`].
+    /// validated [`Configuration`].
     ///
     /// The [`ParseError`] type is provided as a convenience to connector authors, to be used on
     /// error.
     async fn parse_configuration(
         &self,
-        configuration_dir: impl AsRef<Path> + Send,
+        configuration_dir: &Path,
```

Comment:
This change allows `ConnectorSetup` to be "object-safe", which means we can put it in a `Box<dyn ConnectorSetup>`.

Reply:
Aha! How did you find out? was there a clue in the error message?

I suppose this restriction actually makes sense since `f(arg: impl T<X>)` actually becomes `f<Y>(arg: Y) where Y impl T<X>`, and if generics are compiled using specialization/erasure rather than passing around a dictionary of trait methods.

Reply:
I stared it, thought "worth a try", tried it, and it worked! 😛